res/controllers/Pioneer-DDJ-SR2-scripts.js

```diff
@@ -0,0 +1,1411 @@
+
+DDJSR2.init = function() {
+    // Browser section is part of Mixer container for now.
+
+    DDJSR2.browser = new DDJSR2.BrowserContainer();
+
+    DDJSR2.deck = [];
+    for (let i = 0; i < 4; i++) {
+        DDJSR2.deck[i] = new DDJSR2.Deck(i);
+        DDJSR2.deck[i].setCurrentDeck(`[Channel${i+1}]`);
+    }
+
+    DDJSR2.mixer = new DDJSR2.MixerContainer();
+
+    DDJSR2.createEffectUnits();
+
+    // Performance pads are part of the Deck specs, as are Illumination Controls
+
+    DDJSR2.seratoHeartbeatTimer = engine.beginTimer(200, DDJSR2.doSeratoHeartbeatTimer, false);
+
+    // After 500 ms have elapsed, after the serato heartbeat has occurred, then poll controls. Calling this before time results in an unsuccessful poll.
+
+    engine.beginTimer(300, function() {
+        midi.sendSysexMsg(DDJSR2.seratoControlsPoll, DDJSR2.seratoControlsPoll.length);
+    }, true);
+
+};
+
+DDJSR2.doSeratoHeartbeatTimer = function() {
+    midi.sendSysexMsg(DDJSR2.seratoHeartbeat, DDJSR2.seratoHeartbeat.length);
+};
+
+
+// First, a function library.
+////////////////
+
+
+DDJSR2.getRotaryDelta = function(value) {
+    let delta = 0x40 - Math.abs(0x40 - value);
+    const isCounterClockwise = value > 0x40;
+    // BrowserKnob returns 1-30 going down, and 127-98 going up.
+
+    if (isCounterClockwise) {
+        delta *= -1;
+    }
+    return delta;
+};
+
+DDJSR2.getJogWheelDelta = function(value) {
+    // The Wheel control centers on 0x40; find out how much it's moved by.
+    return value - 0x40;
+};
+
+DDJSR2.wheelLedControl = function(channelOffset, color) {
+    const wheelLedBaseChannel = 0xBB;
+    midi.sendShortMsg(
+        wheelLedBaseChannel,
+        0x20+(channelOffset),
+        color
+    );
+};
+
+
+DDJSR2.pitchBendFromJog = function(group, movement) {
+    engine.setValue(group, "jog", movement / 5 * DDJSR2.jogwheelSensitivity);
+};
+
+DDJSR2.RingBufferView = class {
+    constructor(indexable, startIndex = 0) {
+        this.indexable = indexable;
+        this.index = startIndex;
+    }
+    advanceBy(n) {
+        this.index = script.posMod(this.index + n, this.indexable.length);
+        return this.current();
+    }
+    next() {
+        return this.advanceBy(1);
+    }
+    previous() {
+        return this.advanceBy(-1);
+    }
+    current() {
+        return this.indexable[this.index];
+    }
+};
+
+DDJSR2.BrowserContainer = function() {
+    this.browseKnob  = new components.ComponentContainer({
+        longPressTimer: 0,
+        longPressTimeout: 250,
+        trackColorCycleEnabled: false,
+        trackColorCycleHappened: false,
+        previewSeekEnabled: false,
+        previewSeekHappened: false,
+        turn: new components.Encoder({
+            group: "[Library]",
+            input: function(channel, control, value, _status, _group) {
+                const rotateValue = DDJSR2.getRotaryDelta(value);
+                engine.setValue("[Library]", "MoveVertical", rotateValue);
+            }
```

Comment:
pretty sure this works just as well. 
```suggestion
            inKey: "MoveVertical",
            inValueScale : DDJSR2.getRotaryDelta
```